Feature/update docs and example code

dominicschaff opened the pull request:
# Why this change is needed
> Improving the example code and documentation, and start with using the new kubernetes service accounts.

NikitaGandhi commented on examples/stage/03_odc_apps_k8s/main.tf:
```diff
@@ -1,7 +1,7 @@
 terraform {
   backend "s3" {
-    bucket = "odc-test-stage-backend-tfstate"
+    bucket = "odc-test-devtest-backend-tfstate"
```
We cannot assign locals or variables here - https://www.terraform.io/docs/language/settings/backends/configuration.html

> A backend block cannot refer to named values (like input variables, locals, or data source attributes).

woodcockr replied:
Ahh yes, in my haste I noticed the bucket name only and not the context.

dominicschaff commented on examples/stage/01_odc_eks/provider.tf:
```diff
@@ -1,4 +1,18 @@
 provider "aws" {
-  region      = local.region
+  region                 = local.region
```
@NikitaGandhi I am wondering whether it might be a good idea to introduce a variables.tf section in each stage so that the region, state bucket and state files can be supplied there instead of in different files like it is now. Then we won't need the local variables like above

dominicschaff replied:
What would be the best alternative? I would have thought perhaps having a variables section could remove this issue, and it could also be used to store the state file information in as well, however that isn't possible either

dominicschaff commented:
> * skip_region_validation

I have updated the use of locals in the providers, changed the regions to match, made the cognito and cloudfront to be what they were, and removed the skip check

woodcockr commented on examples/stage/03_odc_apps_k8s/efs_provisioner.tf:
```diff
@@ -41,35 +43,36 @@ resource "aws_security_group" "efs" {
 
 data "template_file" "efs_provisioner" {
-  template = file("${path.module}/config/efs-provisioner.yaml")
+  template = file("${path.module}/config/efs_provisioner.yaml")
```
Recommend adding a deprecation notice to the file and a pointer to the new drive documentation.

NikitaGandhi commented on examples/stage/03_odc_apps_k8s/efs_provisioner.tf:
```diff
@@ -0,0 +1,116 @@
+# WARNING: This setup is using deprecated helm chart: https://github.com/kubernetes/charts/tree/master/stable/efs-provisioner
+#   Consider using the newer efs csi driver - https://github.com/kubernetes-sigs/aws-efs-csi-driver
+
```
@woodcockr added warning

mmochan reviewed (approved):
LGTM!

dominicschaff commented:
@woodcockr Are there any further changes that are needed for this to PR to continue?

woodcockr commented:
@dominicschaff Looks like everything I spotted is addressed so all good I believe.
